Explicitly include some header files

joankaradimov opened the pull request:
When compiling master on Windows 10 with MSYS2. I get the following errors:
- error: ‘swab’ was not declared in this scope
- error: ‘strncasecmp’ was not declared in this scope; did you mean ‘strncmp’?

I've added explicit includes according to the documentation for [swab](http://man7.org/linux/man-pages/man3/swab.3.html) and [strcasecmp](http://man7.org/linux/man-pages/man3/strcasecmp.3.html). I assume this compiles on Unices and on GNU because of some transitive includes.

LibRaw commented:
Is there any confidence, that adding that lines will NOT break other compilers? It looks like your additions should be protected by MSYS2-specific #ifdef-s

joankaradimov commented:
I have not tested it in other environments. But note that the documentation links I provided are under `http://man7.org/linux`. And both claim compatibility with `4.4BSD`/`POSIX.1-2001`/`POSIX.1-2008`.

I assume compatibility with `gcc`/`clang` as given. I can test `msvc` later today.

LibRaw commented:
There are a lot of other OSes/compilers. Not sure it is possible to test all, so it is better to localize your change.

Could you please make your change MSYS2-only specific?

joankaradimov commented:
Point taken. I'm localizing my changes. But they have little to do with MSYS2.

I'm suggest including `strings.h` for `strncasecmp` on any non-windows platform. This is where this function is. It works now because of transitive includes. I've put the include in the `#else` of `#if defined WIN32 || defined(__MINGW32__)`. 

I've reverted and squashed my changes that have anything to do with `swab`. I'll try to think of something portable.

LibRaw commented:
LibRaw compiles under Linux and FreeBSD (gcc/clang) without strings.h, so strncasecmp is already here.
It is worth to localize your changes to  compilers/platforms you experience the problem, doing something better may break on some (other, untested) platform.

joankaradimov commented:
> LibRaw compiles under Linux and FreeBSD (gcc/clang) without strings.h, so strncasecmp is already here.

Yes, it is. But it's because of some random transitive include of `strings.h`.

Anyway, this is as good of a fix as I can suggest. Close it if you don't find it useful.

Thank you for your time!
